book/src/building/swift-package/README.md

```diff
@@ -0,0 +1,258 @@
+mkdir macos
+cp ../my-rust-lib/target/x86_64-apple-darwin/debug/libmy_rust_lib.a ./macos
+mkdir simulator
+cp ../my-rust-lib/target/x86_64-apple-ios/debug/libmy_rust_lib.a ./simulator
+```
+
+This should result in the follwing folder structure:
+```
+MyFramework
+├── include
+│   ├── SwiftBridgeCore.h
+│   ├── module.modulemap
+│   └── my_rust_lib.h
+├── ios
+│   └── libmy_rust_lib.a
+├── macos
+│   └── libmy_rust_lib.a
+└── simulator
+    └── libmy_rust_lib.a
+```
+
+Edit `include/module.modulemap`:
+
+```modulemap
+module MyRustLib {
+    header "my_rust_lib.h"
+    header "SwiftBridgeCore.h"
+    export *
+}
+```
+
+Now it is time to build the xcframework:
+
+```bash
+xcodebuild -create-xcframework \
+    -library simulator/libmy_rust_lib.a \
+    -headers include \
+    -library ios/libmy_rust_lib.a \
+    -headers include \
+    -library macos/libmy_rust_lib.a \
+    -headers include \
+    -output MyRustLib.xcframework
+```
+
+*The order of the `library` tags is important, but we don't currently know why*
+
+## Creating the Swift package
+
+Go back to the root of the project (`cd ..`) and create a new Swift package:
+
+```bash
+mkdir MySwiftPackage && cd MySwiftPackage
+```
+
+Now either do `mkdir -r Sources/MySwiftPackage` and `touch Package.swift`, or use `swift package init --type library`.
+
+Copy the xcframework and generated swift files to this folder:
+```bash
+cp -r ../MyFramework/MyRustLib.xcframework ./
+cp ../my-rust-lib/generated/SwiftBridgeCore.swift Sources/MySwiftPackage
+cp ../my-rust-lib/generated/my_rust_lib/my_rust_lib.swift Sources/MySwiftPackage
+```
+
+The folder structure should be:
+```
+MySwiftPackage
+├── Sources
+│   └── MySwiftPackage
+│       └── SwiftBridgeCore.swift
+│       └── my_rust_lib.swift
+├── MyRustLib.b.xcframework
+└── Package.swift
+```
+
+Add the framework as a binary target to `Package.swift`:
+```swift
+// MySwiftPackage/Package.swift
+
+// swift-tools-version:5.5.0
+import PackageDescription
+let package = Package(
+    name: "MySwiftPackage",
+    products: [
+        .library(
+            name: "MySwiftPackage",
+            targets: ["MySwiftPackage"]),
+    ],
+    dependencies: [
+
+    ],
+    targets: [
+        .binaryTarget(
+            name: "MyRustLib",
+            path: "MyRustLib.xcframework"
+        ),
+        .target(
+            name: "MySwiftPackge",
+            dependencies: ["MyRustLib"]),
+    ]
+)
+```
+
+<!--TODO: better way of dealing with this instead of editing the generated files?-->
+We will need to import our rust library in `my_rust_lib.swift` and `SwiftBridgeCore.swift`:
+
+```swift
+// MySwiftPackage/Sources/MySwiftPackage/SwiftBridgeCore.swift
+import MyRustLib
+```
+
+<!--TODO: always make public, or maybe just have an option for making functions public in build.rs?-->
+```swift
+// MySwiftPackage/Sources/MySwiftPackage/my_rust_lib.swift
+import MyRustLib
+
+// Also make the functions here public
+public hello_rust() -> RustString {
```

Comment:
I have asked the user to make the functions public for now. I don't know if we should make SwiftBridge automatically make these public, though. Maybe we have a config option or env variable to opt in to public functions?